addons/binding/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/internal/command/LegacyLiveDataUpdate.java

```diff
@@ -0,0 +1,147 @@
+
+import static org.openhab.binding.solaredge.SolarEdgeBindingConstants.*;
+
+import java.math.RoundingMode;
+import java.nio.charset.StandardCharsets;
+import java.text.DecimalFormat;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.api.Result;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.openhab.binding.solaredge.handler.SolarEdgeHandler;
+import org.openhab.binding.solaredge.internal.callback.AbstractCommandCallback;
+import org.openhab.binding.solaredge.internal.model.LiveDataChannels;
+
+/**
+ * command that retrieves values for live data channels - legacy version for old setups
+ *
+ * @author Alexander Friese - initial contribution
+ *
+ */
+public class LegacyLiveDataUpdate extends AbstractCommandCallback implements SolarEdgeCommand {
+    private static final String NCG_NON_GREEDY_TEXT_OR_NUMBER = "(?:.+?)";
+    private static final String CG_CURRENT_POWER_AND_UNIT = "(?:\\{currentPower:\"([0-9\\.,]+)(.+)\"\\})";
+    private static final String UNIT_W = "W";
+
+    private final SolarEdgeHandler handler;
+    private int retries = 0;
+
+    public LegacyLiveDataUpdate(SolarEdgeHandler handler) {
+        super(handler.getConfiguration());
+        this.handler = handler;
+    }
+
+    @Override
+    protected Request prepareRequest(Request requestToPrepare) {
+        requestToPrepare.followRedirects(false);
+        requestToPrepare.param(DATA_API_LEGACY_LIVE_DATA_FIELDID_FIELD, config.getSolarId());
+        requestToPrepare.method(HttpMethod.GET);
+
+        return requestToPrepare;
+    }
+
+    @Override
+    protected String getURL() {
+        return DATA_API_LEGACY_LIVE_DATA_URL;
+    }
+
+    @Override
+    public void onComplete(Result result) {
+        logger.debug("onComplete()");
+
+        if (!HttpStatus.Code.OK.equals(getCommunicationStatus().getHttpCode()) && retries++ < MAX_RETRIES) {
+            if (getListener() != null) {
+                getListener().update(getCommunicationStatus());
+            }
+            handler.getWebInterface().executeCommand(this);
+
+        } else {
+
+            String html = getContentAsString(StandardCharsets.UTF_8);
+            if (html != null) {
+                handler.updateChannelStatus(parseLegacyData(html));
+            }
+        }
+    }
+
+    /**
+     * uses regexp matcher to extract current production data
+     *
+     * @param data - raw data retrieved from solaredge
+     * @return map containing the live data (only one element)
+     */
+    private Map<String, String> parseLegacyData(String data) {
+        Map<String, String> resultMap = new HashMap<>(1);
+
+        String rawData = data.replaceAll("\\s+", "");
+
+        logger.debug("RAW String: {}", rawData);
+
+        Pattern pattern = compile(NCG_NON_GREEDY_TEXT_OR_NUMBER, CG_CURRENT_POWER_AND_UNIT,
```

Comment:
can we reuse this pattern, regexes are expensive to construct so re-use should be promoted

Reply:
changed it to a final instance member